sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/src/Listeners/ServiceBusScaleMonitor.cs

```diff
@@ -57,23 +58,30 @@ async Task<ScaleMetrics> IScaleMonitor.GetMetricsAsync()
 
         public async Task<ServiceBusTriggerMetrics> GetMetricsAsync()
         {
-            ServiceBusReceivedMessage message = null;
+            ServiceBusReceivedMessage activeMessage = null;
             string entityName = _serviceBusEntityType == ServiceBusEntityType.Queue ? "queue" : "topic";
 
             try
             {
-                // Peek the first message in the queue without removing it from the queue
-                // PeekAsync remembers the sequence number of the last message, so the second call returns the second message instead of the first one
-                // Use PeekBySequenceNumberAsync with fromSequenceNumber = 0 to always get the first available message
-                message = await _receiver.Value.PeekMessageAsync(fromSequenceNumber: 0).ConfigureAwait(false);
+                var receivedMessages = await _receiver.Value.PeekMessagesAsync(10, fromSequenceNumber: lastSequenceNumber).ConfigureAwait(false);
+                foreach (var receivedMessage in receivedMessages)
+                {
+                    if (MessageWasNotScheduledOrDeferred(receivedMessage))
+                    {
+                        activeMessage = receivedMessage;
+                        break;
+                    }
+
+                    lastSequenceNumber = receivedMessage.SequenceNumber;
```

Comment:
Say there are 3 scheduled messages with sequence numbers 1, 2, and 3, and we skip them, setting lastSequenceNumber=3. Won't the above logic mean we won't ever peek 1 or 2 when they become active, since we start at 3?

Reply:
I'll leave addressing this point until the discussion above about how the proceed with this PR is settled OK?

Reply:
They will get new sequence numbers once they move from scheduled to active, so this shouldn't be a problem.

Reply:
https://docs.microsoft.com/en-us/azure/service-bus-messaging/message-sequencing#scheduled-messages

Reply:
Thanks for the clarification